core/include/prometheus/family.h

```diff
@@ -90,6 +91,13 @@ class Family : public Collectable {
   Family(const std::string& name, const std::string& help,
          const std::map<std::string, std::string>& constant_labels);
 
+  bool IsSameAs(
```

Comment:
We must only compare the name. There must only be one TYPE field per family:

> Only one TYPE line may exist for a given metric name.

That means that we need to either consolidate families with the same name, but different help messages/labels or error out.

[1] https://github.com/prometheus/docs/blob/master/content/docs/instrumenting/exposition_formats.md#text-format-details

Reply:
I added a check that the same family name does not show up with a different type. I'd like to work on the other should's and may's in a separate PR.

Reply:
ok